akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -131,20 +163,20 @@ object ActorTestKit {
  *
  * For synchronous testing of a `Behavior` see [[BehaviorTestKit]]
  */
-final class ActorTestKit private[akka] (val name: String, val config: Config, settings: Option[TestKitSettings]) {
+final class ActorTestKit private[akka] (
+    val internalSystem: ActorSystem[_],
+    internalTestKitGuardian: ActorRef[ActorTestKitGuardian.TestKitCommand],
+    settings: Option[TestKitSettings]) {
```

Comment:
`val name` and `val config` are now gone. For compatibility reasons you can add
```
  val name = internalSystem.name
  val config = internalSystem.settings.config
```

I think that should also get rid of the two of the binary compatibility warnings.

The other warnings can be filtered out by adding the following to new file `akka-actor-testkit-typed/src/main/mima-filters/2.6.4.backwards.excludes/27338-ActorTestKit-system.excludes`

```
# #27338 allow passing ActorSystem to ActorTestKit
ProblemFilters.exclude[IncompatibleSignatureProblem]("akka.actor.testkit.typed.scaladsl.ActorTestKit.internalSystem")
ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.actor.testkit.typed.scaladsl.ActorTestKit.this")
```